irida_sistr_results/irida_api.py

```diff
@@ -107,49 +105,47 @@ def get_sistr_results_for_project(self, project):
 		for sample in sistr_results_for_project:
 			sample_pairs=self.irida_connector.get_resources('/api/samples/'+sample['identifier']+'/pairs')
 
-			if len(sample_pairs) == 0:
-				sistr_info = SampleSistrInfo({'sample': sample,
-						'has_results': False
-						})
-			else:
-				sistr_info = None
+			sistr_info = SampleSistrInfo.create_empty_info(sample)
+
+			if len(sample_pairs) > 0:
 				for sequencing_object in sample_pairs:
 					if (self._has_rel_in_links('analysis/sistr', sequencing_object['links'])):
 						sistr_rel=self._get_rel_from_links('analysis/sistr', sequencing_object['links'])
-	
 						sistr=self.irida_connector.get(sistr_rel)
-						
+
 						if (sistr['analysisState'] != 'COMPLETED'):
-							logger.debug("Skipping automated SISTR results associated with sample="+sample['sampleName']+" as state is not 'COMPLETED'.")
-							if sistr_info is None:
-								sistr_info = SampleSistrInfo({'sample': sample,
-									'paired_files': sequencing_object,
-									'has_results': False
-									})
+							logger.debug(
+								"Skipping automated SISTR results associated with sample=%s as state is not 'COMPLETED'.",
+								sample['identifier'])
 						else:
-							sistr_info_curr=self.get_sistr_info_from_submission(sistr)
-							if (sistr_info is None):
-								sistr_info = sistr_info_curr
-							elif sistr_info_curr.get_qc_status() == 'PASS' and (not sistr_info.has_sistr_results() or sistr_info.get_qc_status() != 'PASS'):
-								sistr_info = sistr_info_curr
-							elif sistr_info_curr.get_qc_status() == 'PASS' and (sistr_info_curr.get_submission_created_date() > sistr_info.get_submission_created_date()):
-								sistr_info = sistr_info_curr
-					elif sistr_info is None:
-						sistr_info = SampleSistrInfo({'sample': sample,
-								'paired_files': sequencing_object,
-								'has_results': False
-								})
+							new_sistr_info = self.get_sistr_info_from_submission(sistr)
+							sistr_info = self._update_sample_sistr_info(sistr_info, new_sistr_info, sistr_workflow_id)
 
 			sistr_results.append(sistr_info)
 		return sistr_results
+
+	def _update_sample_sistr_info(self, curr_sistr_info, new_sistr_info, sistr_workflow_id):
```

Comment:
Producing an output of all SISTR results would require quite a bit of re-factoring. In particular a lot of the data structures I use to store results and print the Excel file in the end assumes that one sample maps to one SISTR result. I have added an issue for this https://github.com/phac-nml/irida-sistr-results/issues/5 but I don't think I want to include this functionality in this merge.  What do you think?

For using this method in `irida_sistr_results.IridaSistrResults#_load_additional_sistr_results`, these two have slightly different purposes. `_update_sample_sistr_info` is used only when updating automated analysis, and so includes the quality status in the checks. `_load_additional_sistr_results` is used when loading results shared to a project, and does not include the quality status (so that the more recent result always gets selected).  I describe a bit more of the purpose behind these two pieces of code in the response to the QC rules comment.

Reply:
Making a new PR for outputting all SISTR results would probably be best as it does sound like it would touch a lot of code. 

Thanks for clarifying on the different rules for selecting one set of results over others.